weed/s3api/s3api_objects_list_handlers.go

```diff
@@ -154,7 +154,9 @@ func (s3a *S3ApiServer) listFilerEntries(bucket string, originalPrefix string, m
 						Prefix: fmt.Sprintf("%s/%s/", dir, entry.Name)[len(bucketPrefix):],
 					})
 				}
-				return
+				if entry.Attributes.Mime == "" || !strings.HasSuffix(entry.Name, "/") {
```

Comment:
You may want to put `entry.Attributes.Mime == ""` into some function, add some document, or maybe directly check the mime value, and reuse it in other places.